src/Client/Updater.php

```diff
@@ -185,6 +185,25 @@ public function refresh() : bool
         // TUF-SPEC-v1.0.9 Section 5.3.5
         $this->durableStorage['snapshot.json'] = $newSnapshotContents;
 
+        // TUF-SPEC-v1.0.9 Section 5.4
+        if ($rootData->supportsConsistentSnapshots()) {
+            $targetsVersion = $newSnapshotData->getFileMetaInfo('targets.json')['version'];
+            $newTargetsContent = $this->repoFileFetcher->fetchFile(
+                "$targetsVersion.targets.json",
+                static::MAXIMUM_DOWNLOAD_BYTES
+            );
+            $newTargetsData = TargetsMetadata::createFromJson($newTargetsContent);
+            // TUF-SPEC-v1.0.9 Section 5.4.1
+            $newSnapshotData->verifyNewMetaData($newTargetsData);
```

Comment:
+1 on this thought. Should we also consider a tainting model (where the taint gets removed on verification and accessing tainted data requires acknowledgement that it's not known to be safe)? Maybe that's overkill, though.

Reply:
@davidstrauss ok I added `isTrusted()` for certain getters and added an optional parameter `$allowUntrustedAccess` when sometimes you would need to access this information before the metadata was verified. 

For getters like `getSigned()` and `getVersion()` I didn't add this because we always need to access this before it is signed.

cc @phenaproxima

Reply:
Very cool!